.travis/maybe-skip-build.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+git branch -v
+echo "TRAVIS_COMMIT: $TRAVIS_COMMIT"
+echo "TRAVIS_COMMIT_RANGE: $TRAVIS_COMMIT_RANGE"
+echo "TRAVIS_PULL_REQUEST: $TRAVIS_PULL_REQUEST"
+
+if test -z "$TRAVIS_PULL_REQUEST"; then
+    msg="$TRAVIS_COMMIT_MESSAGE"
+else
+    msg="$(echo "$TRAVIS_COMMIT_RANGE" | cut -d. -f4 |xargs git log --format=%B -n 1)"
+fi
+
+echo "Checking to see whether to skip build for $TARGET_MACHINE due to commit message"
+echo "$msg"
+msg_only="$(echo "$msg" | grep "^travis:only:")"
```

Comment:
I haven’t tried it with an empty list of machine types, but I expect it will skip all of them. It still has to spin up the images and run far enough to execute this code, though. There’s some other magic token you can use to completely skip all CI - something like ci-skip or skip-ci, I think.